ui/src/app/pages/profile/profile-page.tsx

```diff
@@ -101,18 +139,51 @@ export const ProfilePage = withUserProfile()(class extends React.Component<
     super(props);
 
     this.state = {
-      currentProfile: props.profileState.profile || this.createInitialProfile(),
+      currentProfile: this.initializeProfile(),
+      institutions: [],
       saveProfileErrorResponse: null,
       showDemographicSurveyModal: false,
+      showOtherRole: this.initializeShowOtherRole(),
       updating: false
     };
   }
 
+  async componentDidMount() {
+    try {
+      const details = await institutionApi().getPublicInstitutionDetails();
+      this.setState({
+        institutions: details.institutions
+      });
+    } catch (e) {
+      reportError(e);
+    }
+  }
+
+  initializeShowOtherRole() {
+    return this.props.profileState.profile.verifiedInstitutionalAffiliation.institutionalRoleEnum === InstitutionalRole.OTHER;
+  }
+
   navigateToTraining(): void {
     window.location.assign(
       environment.trainingUrl + '/static/data-researcher.html?saml=on');
   }
 
+  initializeProfile() {
+    if (!this.props.profileState.profile) {
+      return this.createInitialProfile();
+    }
+    if (!this.props.profileState.profile.address) {
```

Comment:
Can this be moved into `createInitialProfile`?

Reply:
Its mainly targeting the case where the researchers have old account and therefore do not have the address object.

Reply:
What about merging the two functions, then?